spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -177,6 +184,52 @@ dataflow:> stream deploy --name ticktock --properties "app.time.count=3"
 
 IMPORTANT: See <<spring-cloud-dataflow-stream-app-labels>>.
 
+==== Passing stream partition properties
+A common pattern in stream processing is to partition the data as it is streamed. This entails deploying multiple instances of a message consuming app and using content-based routing so that messages containing the identical
+data value(s) are always routed to the same app instance. You can use the Deployment Properties to declaratively configure a partitioning strategy to route each message to a specific consumer instance.
+
+See below for examples of deploying partitioned streams:
+
+app.[app/label name].producer.partitionKeyExtractorClass::
+  The class name of a PartitionKeyExtractorStrategy (default null)
+
+app.[app/label name].producer.partitionKeyExpression::
+  A SpEL expression, evaluated against the message, to determine the partition key; only applies if partitionKeyExtractorClass is null. If both are null, the app is not partitioned (default null)
+
+app.[app/label name].producer.partitionSelectorClass::
+  The class name of a PartitionSelectorStrategy (default null)
+
+app.[app/label name].producer.partitionSelectorExpression::
+  A SpEL expression, evaluated against the partition key, to determine the partition index to which the message will be routed. The final partition index will be the return value (an integer) modulo [nextModule].count If both the class and expression are null, the underlying binder's default PartitionSelectorStrategy will be applied to the key (default null)
+In summary, an app is partitioned if its count is > 1 and the previous app has a partitionKeyExtractorClass or partitionKeyExpression (class takes precedence). When a partition key is extracted, the partitioned app instance is determined by invoking the partitionSelectorClass, if present, or the partitionSelectorExpression % partitionCount ,
+where partitionCount is count in the case of Redis and RabbitMQ, and the underlying partition count of the topic in the case of Kafka. If neither a partitionSelectorClass nor a partitionSelectorExpression is present the result is key.hashCode() % partitionCount.
```

Comment:
I thought we can still add reference to `redis` as a binder since we still have a repo (at least for dev) for it. may be not?